kafka-client/src/test/java/dev/responsive/kafka/integration/ResponsiveSessionStoreIntegrationTest.java

```diff
@@ -0,0 +1,263 @@
+
+    this.admin = admin;
+    final var result = admin.createTopics(
+        List.of(
+            new NewTopic(inputTopic(), Optional.of(1), Optional.empty()),
+            new NewTopic(otherTopic(), Optional.of(1), Optional.empty()),
+            new NewTopic(outputTopic(), Optional.of(1), Optional.empty())
+        )
+    );
+    result.all().get();
+  }
+
+  @Test
+  public void shouldComputeSessionAggregate() throws Exception {
+    // Given:
+    final Duration inactivityGap = Duration.ofSeconds(5);
+    final Duration gracePeriod = Duration.ofSeconds(2);
+    final SessionWindows window =
+        SessionWindows.ofInactivityGapAndGrace(inactivityGap, gracePeriod);
+
+    final Materialized<String, String, SessionStore<Bytes, byte[]>> store =
+        ResponsiveStores.sessionMaterialized(
+            ResponsiveSessionParams.session(name, inactivityGap, gracePeriod)
+        );
+
+    // Start from timestamp of 0L to get predictable results
+    final List<KeyValueTimestamp<String, String>> inputEvents = asList(
+        new KeyValueTimestamp<>("key", "a", 0L),
+        new KeyValueTimestamp<>("key", "b", 6_000L),
+        new KeyValueTimestamp<>("key", "c", 7_000L),
+        new KeyValueTimestamp<>("key1", "d", 8_000L),
+        new KeyValueTimestamp<>("key1", "e", 16_000L),
+        new KeyValueTimestamp<>("key1", "f", 12_000L),
+        new KeyValueTimestamp<>("key1", "g", 7_500L)
+    );
+    final List<KeyValue<Windowed<String>, String>> expectedPeeks = List.of(
+        new KeyValue<>(windowedKey("key", 0, 0), "a"),
+        new KeyValue<>(windowedKey("key", 0, 0), null),
+        new KeyValue<>(windowedKey("key", 0, 6000), "ab"),
+        new KeyValue<>(windowedKey("key", 0, 0), null),
+        new KeyValue<>(windowedKey("key", 0, 6000), null),
+        new KeyValue<>(windowedKey("key", 0, 7000), "abc"),
+        new KeyValue<>(windowedKey("key1", 8000, 8000), "d"),
+        new KeyValue<>(windowedKey("key1", 16_000, 16_000), "e"),
+        new KeyValue<>(windowedKey("key1", 16_000, 16_000), null),
+        new KeyValue<>(windowedKey("key1", 12000, 16_000), "ef")
+    );
+    final CountDownLatch outputLatch = new CountDownLatch(expectedPeeks.size());
+    final List<KeyValue<Windowed<String>, String>> actualPeeks = new ArrayList<>();
+
+    final StreamsBuilder builder = new StreamsBuilder();
+    final KStream<String, String> input = builder.stream(inputTopic());
+    input
+        .groupByKey()
+        .windowedBy(window)
+        .aggregate(() -> "", sessionAggregator(), sessionMerger(), store)
+        .toStream()
+        .peek((k, v) -> actualPeeks.add(new KeyValue<>(k, v)))
+        .peek((k, v) -> outputLatch.countDown())
+        .selectKey((k, v) -> k.key())
+        .to(outputTopic());
+
+    // When:
+    final Map<String, Object> properties = getMutablePropertiesWithStringSerdes();
+    properties.put(STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+    properties.put(APPLICATION_SERVER_CONFIG, "host1:1024");
```

Comment:
I know you just copied this over so it's probably @agavra who I should aim this comment at, but -- just FYI, the `APPLICATION_SERVER_CONFIG` is really only for IQ and giving instances a name that the user can recognize/control. In other words, we definitely don't need to set it in any of our functional tests (IQ == Interactive Queries, a Streams feature for reading the state store contents from outside the app)